### PR TITLE
Get token id in the API

### LINKED ISSUE #503: Add unique identification in the token API
### Is your feature request related to a problem?

As we can create multiple tokens in a chain, we want to provide a way to uniquely identify a token not only based on the genesis address but also with some additional data.

The reason is, clients need an easy and fast way to identify unique assets.

### Describe the solution you'd like

By combining name, symbol, genesis address we can compute a hash and get an ID or digest with all those values.
This will identify the token.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by samuel-uniris**
# Description

Provide an unique identification of a token for genesis address.
This make possible to host multiple tokens on the same chain while being able to identify them.

Fixes #503 

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/archethic_web/graphql_schema/resolver.ex [hunk 1]
```diff
@@ -41,15 +41,40 @@ defmodule ArchethicWeb.GraphQLSchema.Resolver do
...
     with {:ok, {:ok, genesis_address}} <- Task.yield(t1),
-         {:ok, {:ok, transaction_content}} <- Task.yield(t2) do
+         {:ok,
+          {:ok,
+           %{
+             "name" => name,
+             "supply" => supply,
+             "symbol" => symbol,
+             "type" => type,
+             "properties" => properties
```
Comment (Neylix) [inline lib/archethic_web/graphql_schema/resolver.ex:51]: Here properties may not be present and if we create a token without properties it raise a no match error.
I suggest to use something like `Map.get(map, "properties", []) to return an empty list by default

File: lib/archethic_web/graphql_schema/resolver.ex [hunk 2]
```diff
@@ -41,15 +41,40 @@ defmodule ArchethicWeb.GraphQLSchema.Resolver do
...
+           }}} <- Task.yield(t2) do
+      data_to_digest =
+        case type do
+          "fungible" ->
```
Comment (Neylix) [inline lib/archethic_web/graphql_schema/resolver.ex:55]: I think we should do in the other way, add the properties in the hash of fungible tokens and not in the non-fungible token as non_fungible token will probably have different properties

File: lib/archethic_web/graphql_schema/transaction_type.ex [hunk 3]
```diff
@@ -239,7 +239,8 @@ defmodule ArchethicWeb.GraphQLSchema.TransactionType do
     field(:symbol, :string)
     field(:supply, :integer)
     field(:type, :string)
-    field(:properties, list_of(:tokenProperty))
+    field(:properties, list_of(list_of(:tokenProperty)))
+    field(:id, :string)
```
Comment (Neylix) [inline lib/archethic_web/graphql_schema/transaction_type.ex:243]: You should also add the id description in the text above

**Review by Neylix — CHANGES_REQUESTED** [review]

File: lib/archethic_web/graphql_schema/resolver.ex (hunk 1 above)
Comment (samuel-uniris) [reply lib/archethic_web/graphql_schema/resolver.ex:51]: Ok

File: lib/archethic_web/graphql_schema/resolver.ex (hunk 2 above)
Comment (samuel-uniris) [reply lib/archethic_web/graphql_schema/resolver.ex:55]: Oh yes you re right ! I inverted the logic 😅

File: lib/archethic_web/graphql_schema/transaction_type.ex (hunk 3 above)
Comment (samuel-uniris) [reply lib/archethic_web/graphql_schema/transaction_type.ex:243]: Ok

**Review by Neylix — APPROVED** [review]

**Pull request merged**